sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubProducerAsyncClient.java

```diff
@@ -423,15 +424,38 @@ public String getEventHubName() {
         }).collect(Collectors.toList());
 
         return getSendLink(batch.getPartitionId())
-            .flatMap(link -> messages.size() == 1
-                ? link.send(messages.get(0))
-                : link.send(messages));
+            .flatMap(link -> {
+
+                Context userSpanContext = parentContext.get();
+                parentContext.set(traceSendSpan(userSpanContext, link));
+
+                return messages.size() == 1
+                    ? link.send(messages.get(0))
+                    : link.send(messages);
+
+            }).doOnEach(signal -> {
+                if (isTracingEnabled) {
+                    tracerProvider.endSpan(parentContext.get(), signal);
+                }
+            });
+    }
+
+    /**
+     * Method to start a "Azure.EventHubs.send" span and return the created span in context object.
+     *
+     * @param userSpanContext context containing the optional parent span from the user.
+     * @param link the {@link com.azure.core.amqp.implementation.AmqpSendLink}.
+     *
+     * @return the updated {@link com.azure.core.util.Context} object containing the created send span
+     */
+    private Context traceSendSpan(Context userSpanContext, AmqpSendLink link) {
+        Context entityContext = userSpanContext.addData(ENTITY_PATH_KEY, link.getEntityPath());
```

Comment:
Does this need to be a method? it's only called once

Reply:
Yes, it is only called once. But adding the tracing stuff in a method also helps to explain what happens here and keep it neat?

Reply:
Its easy enough to explain in a comment instead of having to create another call frame for it.